From 01974dad6fb73344e2852715ccfe143b3d998cbf Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 2 Oct 2017 19:24:57 +0200
Subject: [PATCH] Bug 781645 - Skip GSettings schema compile when DESTDIR is
 set

(cherry picked from commit bdc3319e906479e665a9fb30fde084ecf735fa4f)
---
 cmake/modules/GLibTools.cmake | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/cmake/modules/GLibTools.cmake b/cmake/modules/GLibTools.cmake
index fc8eb82..f0be558 100644
--- a/cmake/modules/GLibTools.cmake
+++ b/cmake/modules/GLibTools.cmake
@@ -238,9 +238,11 @@ macro(add_gsettings_schemas _target _schema0)
 	if(_install_code)
 		# Compile gsettings schemas and ensure that all of them are in the place.
 		install(CODE
-			"execute_process(${_install_code}
-				COMMAND ${CMAKE_COMMAND} -E chdir . \"${GLIB_COMPILE_SCHEMAS}\" \"${GSETTINGS_SCHEMAS_DIR}\"
-			)")
+			"if(\"\$ENV{DESTDIR}\" STREQUAL \"\")
+				execute_process(${_install_code}
+					COMMAND ${CMAKE_COMMAND} -E chdir . \"${GLIB_COMPILE_SCHEMAS}\" \"${GSETTINGS_SCHEMAS_DIR}\"
+				)
+			endif(\"\$ENV{DESTDIR}\" STREQUAL \"\")")
 	endif(_install_code)
 endmacro(add_gsettings_schemas)
 
-- 
2.15.1

